Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM python:3.8.6-alpine
+
+## Prepare docker base dependency
+RUN set -euxo pipefail; \
+    apk update; \
+    apk add --no-cache \
+        tini \
+        su-exec
+
+## Create user with home = work directory
+RUN set -euxo pipefail; \
+    addgroup --gid 1001 free-stuff-matrixbot; \
+    adduser --home '/opt/free-stuff-matrixbot' --gecos '' --ingroup free-stuff-matrixbot --disabled-password --uid 1001 free-stuff-matrixbot
+WORKDIR /opt/free-stuff-matrixbot
+
+## Add entrypoint
+COPY ["docker-entrypoint.sh", "/usr/local/bin/docker-entrypoint.sh"]
+ENTRYPOINT ["/sbin/tini", "--", "/usr/local/bin/docker-entrypoint.sh"]
```

Comment:
I always put `ENTRYPOINT` to the end of the Dockerfile, together with `CMD` (if I use both), since they work together when starting the container.

Reply:
I try to sort the Dockerfile instructions by how often they change. And everything below will change far more often than the entrypoint in my experience :man_shrugging: Would you say otherwise? Then I'll change it as per your suggestion.

Reply:
I'd say that `ENTRYPOINT` and `CMD` probably change together too then.
Plus, changing `docker-entrypoint.sh` doesn't result in a change in the actual `ENTRYPOINT`, so I'd prefer grouping `ENTRYPOINT` with `CMD` instead of the `COPY` adding the entrypoint script.

I like them together as `CMD` is basically args for `ENTRYPOINT` so I find it odd to scatter them across the file. In the end it comes down to personal preference I guess.

Reply:
OK, true. For some reason I've always put `CMD` to the end, but you don't have to, afaik. And I like the idea of grouping them together. So… move `CMD` up or `ENTRYPOINT` (and it's `COPY`) down?

Reply:
I always put both of them to the end of the file, but for me it doesn't really matter. We can apply your "sort by how often it changes" "algorithm" here if you like. :D

Reply:
:heavy_check_mark: :D
Then I'll go with moving `CMD` up to `ENTRYPOINT` as an experiment :man_shrugging: (never did this before, but this seems correct)